### ISSUE TITLE
Selection with mouse is not working

### ISSUE THREAD

**Issue by florianrusch** [sachaos/viddy#153]
Text selection via mouse is not working.

I'm currently using viddy 1.1.2-VERGEN_IDEMPOTENT_OUTPUT installed via brew on Mac within iterm.

Also the workaround with hitting <kbd>Space</kbd> or <kbd>s</kbd> doesn't work for me. The only thing that is working is doing a <kbd>Right click</kbd> on the string I want to copy and then select `Copy` from the context menu.

**Comment by sachaos** [comment]
@florianrusch Thank you for reporting.

> mac & iterm.

I'm using mac & iterm, but it works fine.

> The only thing that is working is doing a Right click on the string I want to copy and then select Copy from the context menu.

What is the expected behavior?
Can you copy the text by that operation (right click & select copy from the context menu)?

**Comment by BapRx** [comment]
Hello, I have the same issue.
I just tested previous versions and it never worked after the rust rewrite, the last working version for me is `0.4.0`.

The expected behavior was:
- double click a string while viddy is running
- the text is highlighted (selected)
- the text is copied to the clipboard (because iterm is configured as "Copy on select")

An yes it is possible to copy the text from the context menu but only the selected word instead of a custom selection.

**Comment by sachaos** [comment]
@florianrusch @BapRx I've investigated this problem.
I think it is difficult to enable selecting mouse because now viddy is capturing all mouse behavior.
There are 2 solutions.

1. Implement select and copy.
2. Implement copy all text button.

1 is difficult to implement, but 2 is not difficult, it is easy to implement.
How do you think?
The second solution can solve your problems?

**Comment by BapRx** [comment]
Thanks for looking into it, I found a workaround on iTerm2 which would be even easier than solution 2: Hold `option` key then select the text with your mouse. I don't know if the issue is the same on other terminals and if a similar workaround exist.

It's easy enough for me to be a solution without implementing `select and copy`.

**Comment by florianrusch** [comment]
@BapRx thanks for mentioning the workaround. It also works for me 🙂

Of course from a UX perspective it would be intuitive to just click, select and copy via <kbd>CMD</kbd> + <kbd>C</kbd> or <kbd>y</kbd><kbd>y</kbd>. Personally, I wouldn't think that holding the <kbd>ALT</kbd> (option key) could work. But I fully agree with you, @BapRx, that it makes no sense to put in any effort if the user base affected by this problem is very small. Therefore, for me it would also be more of a "nice-2-have".

Many thanks for the investigations, @sachaos & @BapRx.

@sachaos please feel free to close this issue, if you want.

**Comment by sachaos** [comment]
Thank you @BapRx & @florianrusch
I'll close this issue.
I want to think about copy all feature, so I'll create new issue for that.

**Comment by syhol** [comment]
My team just ran into this problem and it cast a very negative light on viddy. The `option` (mac iterm) or `shift` (mac wezterm) workaround works, but it's very unintuitive and it's another thing we need to document and teach new staff.

"Copy all" is not something we would use. We need to suspend execution and copy a line or a word from the output.

We might end up downgrading to 0.3.7 until this is resolved.

Despite this issue we really appreciate everything viddy provides, so thank you for all the work you've put into it @sachaos  🙇

**Comment by sachaos** [comment]
OK, let's reopen this issue.

**Comment by sachaos** [comment]
I would like to add option to disable the mouse control feature firstly.

**Comment by sachaos** [comment]
@syhol Will viddy v1.0.1 meet your requirements? It is a version prior to enabling mouse control.

**Comment by syhol** [comment]
Thank you for reopening the issue. v1.0.1 works nicely. We're looking forward to the disable mouse feature.

**Comment by sachaos** [comment]
@syhol Thank you!
I've released v1.3.0 and added `--disable_mouse` option.
You can configure the default behavior by Configuration https://github.com/sachaos/viddy?tab=readme-ov-file#configuration .

**Comment by syhol** [comment]
Amazing! that works. We manage viddy though mise and the aqua registry so I need to wait for it to trickle through those before I can roll it out to my team, but I've tested it successfully with this:
```console
❯ mise exec ubi:sachaos/viddy@1.3.0 -- viddy --disable_mouse date
```

Thanks for the quick turnaround 🙏
